### ISSUE TITLE
A wrong error message when using embedding arguments keyword name

### ISSUE THREAD

**Issue by Teddy12090** [nokia/RED#179]
The code below shows a test case and a keyword.

```robotframework
*** Test Cases ***
Sample
    Log A
    Log a

*** Keywords ***
Log ${var}
    Log    ${var}
```

When I have the code in RED editor, the IDE shows an error message (see the image below). The message shows **Unknown keyword 'Log a'**. But the keyword used in line 4 is defined in lines 7-8 and the test case can be run correctly, I think it is a wrong error message.

![image](https://user-images.githubusercontent.com/3054877/35712877-5f7a86b8-07ff-11e8-9430-5540d2d228c8.png)

**Comment by michalanglart** [comment]
Looks like a bug, especially given the fact that upper case Log A is recognized.

What is the RED version btw?

**Comment by Teddy12090** [comment]
I think  RED is 0.8.2. But the installation details of RED shows two: 0.8.1 and 0.8.2.

![image](https://user-images.githubusercontent.com/3054877/35728590-0c7c9b84-0847-11e8-9a29-2e47feb0a603.png)

**Comment by michalanglart** [comment]
Looks like you've downloaded product in version 0.8.1 and then upgraded to 0.8.2, so this is the latest version

**Comment by michalanglart** [comment]
I was able to solve this while working on #225

Normally when validation is running it caches analyzed symbols so that for example keyword 'Log' is searched only once for whole file. The cache stores the names in canonical forms (lower case, no underscores, no spaces) however here in one of cases the name was stored as is (`Log A`) thus creating problem when checking `Log a`.

Offending line:
https://github.com/nokia/RED/blob/d0f2b115ca165aa0276ba2eb942d8eb62385da2d/src/RobotFrameworkCore/org.robotframework.ide.core-functions/src/main/java/org/rf/ide/core/testdata/model/search/keyword/KeywordSearcher.java#L94

**Comment by michalanglart** [comment]
Fixed in 0.8.6

**Comment by KrzysztofJozefowicz** [comment]
released in 0.8.7
